docs/architecture/notifications.md

```
@@ -0,0 +1,579 @@
+```
+
+### Notification Discovery
+
+```
+GET    /api/gateways/{gateway}/channels                           -- discovered channels
+GET    /api/gateways/{gateway}/channels/{channel}                 -- channel detail + subscribers
+```
+
+### Agent Subscription Management
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)